common/environment/environment.cpp

```diff
@@ -966,12 +1004,42 @@ void CLocalEnvironment::buildMachineCache() const
                 x.append(name).append("\"]");
                 Owned<IConstEnvBase> cached = new CConstMachineInfo((CLocalEnvironment *) this, &it->query());
                 cache.setValue(x.str(), cached);
+                if (streq(".", name) || strieq("localhost", name) || streq("127.0.0.1", name) || streq(name, localHostIP.str()))
```

Comment:
Strictly speaking anything matching 127.255.255.255 will point to the loopback device, so should really look for that.

Actually you should be using isLoopBack() from jsocket to test this.
Or if not a loopback IP, you should be using isLocal() to test if it's an interfface IP (as opposed to the test against localHostIP), since there can be >1 local interface.